### PR TITLE
Api install

### LINKED ISSUE #99: `api.pkg.Install`
Impelement api.pkg methods for installing, updating and other stuff related to making changes on packages

It doesn't need to be made using VRCT, meaning I won't be mad if you would install package immediately after calling this method

The only thing you need to worry about is situation when dependency A wants to install package which dependency B wants to uninstall
or situation when dependency A wants to install package 1, dependency B wants to install package 2 while packages 1 and 2 are in conflict

### LINKED ISSUE #104: `spito check` doesn't work
This spito.yml
```yml
repo_url: https://github.com/avorty/spito-ruleset
git_prefix: github.com
identifier: avorty/spito-ruleset

rules:
    dbus: ./rules/dbus.lua
    nerd-fonts: ./rules/nerd-fonts.lua
    info: ./rules/info.lua
```

Causes this error when executing `spito g samba`
```
[error] yaml: unmarshal errors:
  line 6: cannot unmarshal !!str `./rules...` into cmd.Rule
  line 7: cannot unmarshal !!str `./rules...` into cmd.Rule
  line 8: cannot unmarshal !!str `./rules...` into cmd.Rule
```

As a result command `spito check avorty/spito-ruleset info` throws this output:
```
Failed to read script called: info from github.com/avorty/spito-ruleset
open /home/jarek/.local/state/spito/rulesets/github.com/avorty/spito-ruleset/spito.yaml: no such file or directory
```
BTW it doesn't end with a new line:
image

### PR DESCRIPTION
**Pull request by sewe2000**
Add ability to install and remove system packages in rules + fix some bugs.
Resolves #99
Closes #104

### CODE REVIEW

File: internal/checker/api_binding.go [hunk 1]
```diff
@@ -10,10 +10,10 @@ import (
...
 // Every cmdApi needs to be attached here in order to be available:
-func attachApi(importLoopData *shared.ImportLoopData, ruleConf *RuleConf, L *lua.LState) {
+func attachApi(importLoopData *shared.ImportLoopData, ruleConf *shared.RuleConfigLayout, L *lua.LState) {
 	apiNamespace := newLuaNamespace()
 
-	apiNamespace.AddField("pkg", getPackageNamespace(L))
+	apiNamespace.AddField("pkg", getPackageNamespace(L, importLoopData))
 	apiNamespace.AddField("sys", getSysInfoNamespace(L))
 	apiNamespace.AddField("fs", getFsNamespace(L, importLoopData))
 	apiNamespace.AddField("info", getInfoNamespace(importLoopData, L))
```
Comment (BaderBC) [inline internal/checker/api_binding.go:19]: Info namespace and Package namespace both takes the same arfuments, but in different order
Please make it consistent (I propose changing Info namespace arguments order)

File: cmd/generate.go [hunk 2]
```diff
@@ -20,49 +21,53 @@ const rulesDirectory = "rules"
 
 func handleGenerate(cmd *cobra.Command, args []string) {
 	rulePath := strings.ReplaceAll(args[0], " ", "")
+	if strings.HasSuffix(rulePath, ".lua") {
+		rulePath = strings.TrimSuffix(rulePath, ".lua")
```
Comment (cooligus) [inline cmd/generate.go:25]: I think you can use `TrimSuffix` without if above

File: pkg/package_conflict/package_conflict_tracker.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	return PackageConflictTracker{
+		packagesInstalled: make(map[string]bool),
+		packagesRemoved:   make(map[string]bool),
+	}
+}
+
+func (packageTracker PackageConflictTracker) AddPackage(packageName string) error {
+
+	if _, isPackageUninstalled := packageTracker.packagesRemoved[packageName]; isPackageUninstalled {
+		return errors.New(fmt.Sprintf("[PACKAGE_CONFLICT] the package %s is required to be uninstalled by a dependency", packageName))
```
Comment (cooligus) [inline pkg/package_conflict/package_conflict_tracker.go:23]:
```suggestion
		return fmt.Errorf("[PACKAGE_CONFLICT] the package %s is required to be uninstalled by a dependency", packageName)
```

**Review by cooligus — CHANGES_REQUESTED** [review]
You can also add Resolves #82

**Review by BaderBC — CHANGES_REQUESTED** [review]
As I previously said - we need to install packages immediately, because sometimes e.g. installing package may override your changes or config which you wanna modify should be created while installing package

**Review by cooligus — APPROVED** [review]

**Comment by sewe2000** [comment]
> As I previously said - we need to install packages immediately, because sometimes e.g. installing package may override your changes or config which you wanna modify should be created while installing package

What do you mean by that?

**Review by cooligus — CHANGES_REQUESTED** [review]

**Review by BaderBC — CHANGES_REQUESTED** [review]
- Implement progress bar in `api.info` then use it automatically when installing package
- Add support for aur packages
- Add tests

**Review by cooligus — APPROVED** [review]
I don't like idea of installing package before applying rule, but...

Edit:
I also realized that installing packages from aur also requires root (issue welcome).

**Review by BaderBC — APPROVED** [review]

**Pull request merged**